elide-datastore/elide-datastore-jpa/src/main/java/com/yahoo/elide/datastores/jpa/transaction/AbstractJpaTransaction.java

```diff
@@ -295,4 +321,45 @@ public Object getRelation(
     public void cancel(RequestScope scope) {
         jpaTransactionCancel.accept(em);
     }
+
+
+    /**
+     * Creates an entity graph for the first encountered to Many relationship that meets the following criteria
+     *  * The relationship has to be present in Entity Projection
+     *  * The relationship type should be ToMany and not computed relationship
+     *  * The relationship Entity should not be presnet in dirtyResource.
+     *      (no mutable operation is done on the relationship entity in this request)
+     *  * If the relationship Entity is presnet in dirtyResource,
+     *      then the relationshipship should have CASCADE.ALL cascade type.
+     * @param scope
+     * @param projection
+     * @return
+     */
+    private EntityGraph createEntityGraph(RequestScope scope, EntityProjection projection) {
+        EntityDictionary dictionary = scope.getDictionary();
+        Class<?> entityClass = projection.getType();
+        List<String> relationshipNames = dictionary.getRelationships(entityClass);
+        for (String relationshipName : relationshipNames) {
+            if (!projection.getIncludedRelationsName().contains(relationshipName)) {
+                continue;
+            }
+            RelationshipType type = dictionary.getRelationshipType(entityClass, relationshipName);
```

Comment:
Rather than looking up this information the dictionary (name and class of the relationship), you can get it from the entity projection.